beacon-chain/sync/initial-sync/blocks_queue.go

```diff
@@ -0,0 +1,413 @@
+package initialsync
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	"github.com/prysmaticlabs/prysm/shared/mathutil"
+)
+
+const (
+	// queueMaxPendingRequests limits how many concurrent fetch request queue can initiate.
+	queueMaxPendingRequests = 8
+	// queueFetchRequestTimeout caps maximum amount of time before fetch requests is cancelled.
+	queueFetchRequestTimeout = 60 * time.Second
```

Comment:
1 minute is quite long to schedule a single request, is there any reason its so long ?

Reply:
Well, that's not for scheduling, scheduling happens (normally) faster. However, when we have problems with peers returning data, we need to give them some time before shutting requests to them down. So, we have `queueMaxPendingRequests = 8` concurrent requests, and each can timeout in 60 seconds. I believe, at least 30 seconds, is definitely what we must give peers to respond (before cancelling request/context), but whether we need that 60 seconds, I am not sure. Please, feel free to suggest better value.

Reply:
So if a peer doesn't respond to us in 15 seconds we automatically close the stream:
https://github.com/prysmaticlabs/prysm/blob/5657535c5253699372974facee6058df9f35b19f/beacon-chain/p2p/sender.go#L24

It might not be necessary to set the timeout to be so long in this case, probably something like 25s for 8 concurrent requests. Although they should all be resolved in 15s. So even that maybe too high.